codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -366,6 +366,27 @@ public HttpHeaders copy() {
     }
 
     private static void validateHeaderNameElement(byte value) {
+        // Check to see if the character is not an ASCII character, or invalid
+        if (value < 0) {
+            throw new IllegalArgumentException("a header name cannot contain non-ASCII character: " + value);
+        }
+        if (!VALID_HEADER_NAME_ELEMENT[value]) {
+            throw new IllegalArgumentException(
+                    "a header name cannot contain the following prohibited characters: =,;: \\t\\r\\n\\v\\f: " +
+                    value);
+        }
+    }
```

Comment:
I think this should be a bit cheaper:

```suggestion
        try {
            if (!VALID_HEADER_NAME_ELEMENT[value]) {
                throw new IllegalArgumentException(
                        "a header name cannot contain the following prohibited characters: =,;: \\t\\r\\n\\v\\f: " +
                        value);
            }
        } catch (IndexOutOfBoundsException e) {
            throw new IllegalArgumentException("a header name cannot contain non-ASCII character: " + value);
        }
    }
```